flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/connectors/hive/util/JobConfUtils.java

```diff
@@ -40,4 +44,24 @@ public static String getDefaultPartitionName(JobConf jobConf) {
                 HiveConf.ConfVars.DEFAULTPARTITIONNAME.varname,
                 HiveConf.ConfVars.DEFAULTPARTITIONNAME.defaultStrVal);
     }
+
+    private static void addCredentialsIntoJobConf(JobConf jobConf) {
+        try {
+            Credentials currentUserCreds =
+                    HadoopInputFormatCommonBase.getCredentialsFromUGI(
+                            UserGroupInformation.getCurrentUser());
+            if (currentUserCreds != null) {
+                jobConf.getCredentials().mergeAll(currentUserCreds);
+            }
+        } catch (Exception e) {
+            throw new RuntimeException(
```

Comment:
I don't think we should error out if `HadoopInputFormatCommonBase.getCredentialsFromUGI` fails, right?

Reply:
`HadoopInputFormatCommonBase.getCredentialsFromUGI` can be rewritten to `return ugi.getCredentials();` since Flink has already dropped support for Hadoop 1.x (https://issues.apache.org/jira/browse/FLINK-4895)

Reply:
Yeah I think we can make that change. Please split it into a separate commit.